test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs

```diff
@@ -40,58 +56,110 @@ public class AddressRegistrationTests
 
             var hostBuilder = new WebHostBuilder()
                 .UseConfiguration(config)
-                .UseKestrel()
+                .UseKestrel(options =>
+                {
+                    options.UseHttps(@"TestResources/testCert.pfx", "testPassword");
+                })
                 .Configure(ConfigureEchoAddress);
 
             using (var host = hostBuilder.Build())
             {
                 host.Start();
 
-                using (var client = new HttpClient())
+                RemoteCertificateValidationCallback validationCallback =
+                        (sender, cert, chain, sslPolicyErrors) => true;
+
+                try
                 {
-                    foreach (var testUrl in testUrls)
+#if NET451
+                    var handler = new HttpClientHandler();
+                    ServicePointManager.ServerCertificateValidationCallback += validationCallback;
+#else
+                    var handler = new WinHttpHandler();
```

Comment:
We shouldn't do this for non-HTTPS addreses. This fails on Linux and OS X (see the Travis runs), so we need to use OSSkipConditions like we do for CanReadAndWriteWithHttpsConnectionFilter to use WinHttpHandler.

I think we should split the RegisterAddresses* Theories into HTTP/HTTPS versions so we don't need to skip more tests than necessary.